### DISCUSSION TITLE
Update SigNoz user guide

### DISCUSSION THREAD

**Discussion by pranay01** [SigNoz/signoz#779] [Ideas]
_This discussion has project proposal for SigNoz's participation in the Google Season of Docs 2022 program_

-----

 **1. Project Title** 
Update SigNoz user guide
 
**2. About SigNoz** 
 SigNoz is an open source observability platform. We help developers monitor applications and troubleshoot any problems they have. The project was made public in February 2021, and since then there have been 50+ contributors to the project with ~6000 github stars. There are around 750+ members in the project slack community.
 
 **3. Project Problem**
Developers want to use SigNoz to monitor their application and troubleshoot problems that may occur while running applications in production. We want to make this process as simple as possible. 

This broadly involves 3 steps:
- Install SigNoz
- Instrument their applications to start sending data to SigNoz
- Visualise their data in SigNoz dashboards and find insights to solve issues

Developers use different kind of infrastructure to install SigNoz ( e.g. Kubernetes, Docker) and have different languages and frameworks in which their applications are written. This makes the surface area of the product very large. The documentation should be written in such a way that users are able to find instructions they would need to be productive with SigNoz very quickly. As mentioned, since there are multiple languages and frameworks, this problem is non-trivial.

Developers when try out new product also want to get started quickly. So the key steps and defaults should be very easily accessible so that users can find the key steps without the need to read the complete docs in details

**4. Project Scope**

The project will:

- Audit the existing documentation and create a friction log of the current documentation for the three top use cases ([Install SigNoz](https://signoz.io/docs/install/) , [Instrument your application](https://signoz.io/docs/instrumentation/), and [using SigNoz](https://signoz.io/docs/userguide/overview/) ).
- Using the friction log as a guide for understanding the gaps in the documentation, create updated documentation for the top use cases.
- Create a quick introductory guide to get users started with the key concepts of observability and distributed tracing
- Incorporate feedback from documentation testers (volunteers in the project) and the wider SigNoz community.
- Work with the release team to update the documentation on the SigNoz website, and to create a process for keeping the documentation in sync with the update tool going forward.

**5. Measuring Project's success**
 SigNoz receives many issues and questions from users in our public slack community, github issues and via emails and chats. Many issues/questions are questions around troubleshooting for their specific installation or how to specific things in SigNoz for their application stack. 

We think a good documentation should help us reduce these questions/issues which could be covered in more detail in our docs. Apart from details it should also be made more accessible so that it can be easiy found by users when they are looking for it.

We would consider the project successful if, after publication of the new documentation:
1. The numbers of issues/questions in our slack and github community related to installing SigNoz and instrumenting applications decreases by at least 15% per month.
2. Number of users able to successfully send data to SigNoz in a month increases by 15%

**Timelines**
Dates | Action Items
-- | --
May | Orientation
June - August | Audit existing documentation and create friction log
September - October | Create documentation
November | Project completion

----

For folks who are interested in sending a proposal - 

Please send your proposal to `gsod@signoz.io`. Add 'GSoD 2022 - SigNoz Proposal' as subject
In the proposal, please attach your prior writing samples as well - preferably focused around technical content

For those interested in discussing about your proposal - join our [slack community](https://signoz.io/slack) and ask in #general channel

**Please send your proposals to above email id by May 1, 2022**

**Comment by vedant-jain03** [comment]
Hello, @pranay01

My self vedant Jain, I am interested in working with Signoz in GSoD this year. How can I contact you to get review my Proposal?

**Reply by pranay01** [reply]
Hi @vedant-jain03 thanks for your interest. Please send your proposal to `gsod@signoz.io` 
In the proposal, please attach your prior writing samples as well - preferably focused around technical content

**Reply by pranay01** [reply]
Please add 'GSoD 2022 - SigNoz Proposal' as subject

**Reply by vedant-jain03** [reply]
Sure, will send the proposal soon.

**Reply by vedant-jain03** [reply]
@pranay01 I have sent the proposal.

**Comment by kelvinj54** [comment]


**Reply by pranay01** [reply]
Hi @kelvinj54  thanks for your interest. Please send your proposal to `gsod@signoz.io` 
In the proposal, please attach your prior writing samples as well - preferably focused around technical content

**Reply by pranay01** [reply]
Please add 'GSoD 2022 - SigNoz Proposal' as subject

**Comment by luphieanza** [comment]
Hello, @pranay01 I'm interested to join as technic writer at SigNoz for GSoD program.

**Reply by luphieanza** [reply]
Hi just want to update you, I've been sending the proposal mail to the gsod@signoz.io please have a take a look!

**Reply by luphieanza** [reply]
@pranay01 are we have a community channel to chat?

**Reply by pranay01** [reply]
https://signoz.io/slack

**Comment by Iheanacho-ai** [comment]
Hey @pranay01, please, I write predominantly javascript and javascript-related frameworks (React, Next.js, Vuejs, Nuxt.js) and docker. Do you think this would be a good project for me

**Reply by pranay01** [reply]
@Iheanacho-ai You can check the current docs and see if you are able to understand it. https://signoz.io/docs

**Comment by imanishbarnwal** [comment]
Hey @pranay01, is the organization still accepting proposals or results are out? Let me know, thank you!!

**Reply by zriyansh** [reply]
SigNoz has already selected a candidate of their choice, the deadline was 1st May.

**Reply by pranay01** [reply]
@imanishbarnwal We have already selected candidate for GSoD.
